src/ddmd/mars.d

```diff
@@ -1519,20 +1555,56 @@ Language changes listed by -transition=id:
 
     // inlineScan incrementally run semantic3 of each expanded functions.
     // So deps file generation should be moved after the inlinig stage.
-    if (global.params.moduleDeps)
+    if (global.params.moduleDeps || global.params.makeDeps)
     {
         foreach (i; 1 .. modules[0].aimports.dim)
             semantic3OnDependencies(modules[0].aimports[i]);
 
-        OutBuffer* ob = global.params.moduleDeps;
-        if (global.params.moduleDepsFile)
+        if (global.params.moduleDeps)
         {
-            auto deps = File(global.params.moduleDepsFile);
-            deps.setbuffer(cast(void*)ob.data, ob.offset);
-            writeFile(Loc(), &deps);
+            OutBuffer* ob = global.params.moduleDeps;
+            if (global.params.moduleDepsFile)
+            {
+                auto deps = File(global.params.moduleDepsFile);
+                deps.setbuffer(cast(void*)ob.data, ob.offset);
+                writeFile(Loc(), &deps);
+            }
+            else
+                printf("%.*s", cast(int)ob.offset, ob.data);
+        }
+        if (global.params.makeDeps)
+        {
+            const(char)* outpath = global.params.makeDepsFile;
+            const(char)* mtarget = global.params.makeDepsTarget;
+            if (!outpath)
+            {
+                outpath = FileName.forceExt((*global.params.objfiles)[0], global.makedep_ext);
+            }
+            OutBuffer* depBuffer = new OutBuffer();
+            if (mtarget)
+                depBuffer.printf("%s: ", mtarget);
+            else
+            {
+                escapePathSpaces(depBuffer, (*global.params.objfiles)[0]);
+                depBuffer.writestring(": ");
+            }
+            foreach (ref mod; modules)
+            {
+                escapePathSpaces(depBuffer, mod.srcfile.toChars());
+                depBuffer.writeByte(' ');
+            }
+            foreach (dep; (*global.params.makeDeps)[])
+            {
+                depBuffer.writeByte('\\');
+                depBuffer.writenl();
+                depBuffer.writeByte(' ');
+                escapePathSpaces(depBuffer, dep);
+                depBuffer.writeByte(' ');
+            }
+            auto depf = File(outpath);
+            depf.setbuffer(cast(void*)depBuffer.data, depBuffer.offset);
```

Comment:
Fixed. This is my first encounter with dmd's code, so sorry that I didn't notice the `_ref` field in File.